### PR TITLE
ci(e2e): Run Tests in Pipelines

### PR DESCRIPTION
**Pull request by peintnermax**
- Provide e2e tests via cypress
- shall be able to be executed with docker compose for isolated test
- shall be able to run during frontend development process
- shall be executed automatically on releases

### CODE REVIEW

**Comment by eliobischof** [comment]
Blocked by #3978

File: build/e2e/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+ARG GO_VERSION=1.17
```
Comment (fforootd) [inline build/e2e/Dockerfile:1]: We should change this to a external variable from GH since it is used in multiple places.

Also we should use a default

Comment (thesephirot) [reply build/e2e/Dockerfile:1]: i am with you, this is still early q&d, I will clean this up. same for versions of products (ZITADEL, COCKROACHDB, etc)

Comment (fforootd) [reply build/e2e/Dockerfile:1]: Yeah my comment is more intended for further tracking 😁

**Comment by eliobischof** [comment]
To be implemented:
- [ ] Merge #4114 
- [ ] Reduce the file e2e/install.sh and rename it to build.sh and print the built image name to stdout
- [ ] Reuse npm modules downloaded by goreleaser
- [ ] Cache docker layers in pipelines

**Comment by thesephirot** [comment]
why do we change files in the cmd directory? they should not be affected by the tests.

**Comment by eliobischof** [comment]
> why do we change files in the cmd directory? they should not be affected by the tests.

Tried to revert all relevant changes there. I'll check it again.

**Review by thesephirot — APPROVED** [review]

**Pull request merged**